pkg/controller/config.go

```diff
@@ -21,4 +21,29 @@ type Config struct {
 	Workers              int
 	Kubeconfig           string
 	MasterURL            string
+	Controllers          []string
+}
+
+// shouldEnableController checks if a specific controller should be enabled
+func (cc Config) shouldEnableController(controllerName string) bool {
+	// If no controllers are specified, enable all by default
+	if len(cc.Controllers) == 0 {
+		return true
+	}
+
+	// Check if the specified controller is in the list
+	for _, ctrl := range cc.Controllers {
+		if ctrl == controllerName {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
we can track all the controllers in a map to facilitate validating and also check whether a specific one has been enabled